internal/cmd/connect/command_plugin_install.go

```
@@ -0,0 +1,456 @@
+	}
+	nameVersionSplit := strings.Split(ownerNameSplit[1], ":")
+	if len(nameVersionSplit) != 2 || nameVersionSplit[0] == "" || nameVersionSplit[1] == "" {
+		return "", "", "", errors.NewErrorWithSuggestions(parsePluginErrorMsg, parsePluginSuggestions)
+	}
+
+	return ownerNameSplit[0], nameVersionSplit[0], nameVersionSplit[1], nil
+}
+
+func getLocalManifest(archivePath string) (*manifest, error) {
+	zipReader, err := zip.OpenReader(archivePath)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to open local archive file %s", archivePath)
+	}
+	defer zipReader.Close()
+
+	for _, zipFile := range zipReader.File {
+		isManifest, err := filepath.Match("*/manifest.json", filepath.ToSlash(zipFile.Name))
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to examine file %s inside local archive file %s", zipFile.Name, archivePath)
+		}
+
+		if isManifest {
+			manifestFile, err := zipFile.Open()
+			if err != nil {
+				return nil, errors.Wrapf(err, "failed to open manifest file %s inside local archive file %s", zipFile.Name, archivePath)
+			}
+			defer manifestFile.Close()
+
+			jsonByteArr, err := io.ReadAll(manifestFile)
+			if err != nil {
+				return nil, err
+			}
+
+			pluginManifest := new(manifest)
+			if err := json.Unmarshal(jsonByteArr, &pluginManifest); err != nil {
+				return nil, err
+			}
+
+			return pluginManifest, nil
+		}
+	}
+
+	return nil, errors.Errorf(`failed to find manifest file inside local archive file "%s"`, archivePath)
+}
+
+func getRemoteManifest(owner, name, version string) (*manifest, error) {
+	manifestUrl := fmt.Sprintf("https://api.hub.confluent.io/api/plugins/%s/%s", owner, name)
+	if version != "latest" {
+		manifestUrl = fmt.Sprintf("%s/versions/%s", manifestUrl, version)
+	}
+
+	r, err := http.Get(manifestUrl)
+	if err != nil {
+		return nil, err
+	}
+
+	if r.StatusCode != http.StatusOK {
+		return nil, errors.New("failed to read manifest file from Confuent Hub")
```

Why not surface the backend error?